pkg/common/api/middleware/middleware.go

```diff
@@ -4,15 +4,17 @@ import (
 	"context"
 )
 
-type PreprocessFunc = func(ctx context.Context, fullMethod string) (context.Context, error)
+type PreprocessFunc = func(ctx context.Context, req interface{}, fullMethod string) (context.Context, error)
 type PostprocessFunc = func(ctx context.Context, fullMethod string, handlerInvoked bool, rpcErr error)
 
 type Middleware interface {
 	// Preprocess is invoked before the gRPC handler is called. It returns a
 	// (possibly modified) context that is passed into the handler, which
 	// should either be the context passed into the function or one derived
 	// from it. If the function returns an error, the gRPC method fails.
-	Preprocess(ctx context.Context, fullMethod string) (context.Context, error)
+	// req passes the request object for unary interceptors and nil for
+	// stream interceptors
+	Preprocess(ctx context.Context, req interface{}, fullMethod string) (context.Context, error)
```

Comment:
The nittiest of all nitpicks :) How would you feel about adding the `req` parameter after `fullMethod`? That has the benefit of keeping the first two arguments the same between `Preprocess` and `Postprocess`.

Reply:
of course :). Done!